instaloader/nodeiterator.py

```diff
@@ -318,12 +319,14 @@ def resumable_iteration(context: InstaloaderContext,
             context.error("Warning: Not resuming from {}: {}".format(resume_file_path, exc))
     try:
         yield is_resuming, start_index
-    except (KeyboardInterrupt, AbortDownloadException):
+    except (KeyboardInterrupt, AbortDownloadException, QueryReturnedBadRequestException):
         if os.path.dirname(resume_file_path):
             os.makedirs(os.path.dirname(resume_file_path), exist_ok=True)
         save(iterator.freeze(), resume_file_path)
         context.log("\nSaved resume information to {}.".format(resume_file_path))
         raise
-    if resume_file_exists:
-        os.unlink(resume_file_path)
-        context.log("Iteration complete, deleted resume information file {}.".format(resume_file_path))
+    finally:
```

Comment:
No, moving this in a finally block is most likely undesired, it causes the resume file to be deleted right after it is updated.